tests/test.c

```diff
@@ -3093,6 +3095,99 @@ void test_consumer_verify_assignment0(const char *func,
         rd_kafka_topic_partition_list_destroy(assignment);
 }
 
+/**
+ * @brief Verify that the consumer's assignment matches the expected assignment.
+ *        passed as a topic partition list in \p expected_assignment .
+ */
+rd_bool_t test_consumer_verify_assignment_topic_partition_list0(
+    const char *func,
+    int line,
+    rd_kafka_t *rk,
+    rd_kafka_topic_partition_list_t *expected_assignment) {
+        rd_kafka_topic_partition_list_t *assignment;
+        rd_kafka_resp_err_t err;
+        int i;
+        rd_bool_t ret = rd_true;
+
+        if ((err = rd_kafka_assignment(rk, &assignment)))
+                TEST_FAIL("%s:%d: Failed to get assignment for %s: %s", func,
+                          line, rd_kafka_name(rk), rd_kafka_err2str(err));
+
+        TEST_SAY("%s assignment (%d partition(s)):\n", rd_kafka_name(rk),
+                 assignment->cnt);
+        for (i = 0; i < assignment->cnt; i++)
+                TEST_SAY(" %s [%" PRId32 "]\n", assignment->elems[i].topic,
+                         assignment->elems[i].partition);
+
+        rd_kafka_topic_partition_list_sort(assignment, NULL, NULL);
+        rd_kafka_topic_partition_list_sort(expected_assignment, NULL, NULL);
+
+        if (assignment->cnt != expected_assignment->cnt) {
+                ret = rd_false;
+                goto done;
+        } else {
+                for (i = 0; i < assignment->cnt; i++) {
+                        if (strcmp(assignment->elems[i].topic,
+                                   expected_assignment->elems[i].topic) ||
+                            assignment->elems[i].partition !=
+                                expected_assignment->elems[i].partition) {
+                                ret = rd_false;
+                                goto done;
+                        }
+                }
+        }
```

Comment:
`goto done` or `break`. Whatever you want.

```suggestion
        if (assignment->cnt != expected_assignment->cnt) {
                ret = rd_false;
                goto done;
        }

        for (i = 0; i < assignment->cnt; i++) {
                if (strcmp(assignment->elems[i].topic,
                            expected_assignment->elems[i].topic) ||
                    assignment->elems[i].partition !=
                        expected_assignment->elems[i].partition) {
                        ret = rd_false;
                        break;
                }
        }
```